packages/react-instantsearch/src/core/createConnector.js

```diff
@@ -139,7 +139,7 @@ export default function createConnector(connectorDesc) {
         metadata,
       } = store.getState();
       const searchState = {results, searching, error};
-      return connectorDesc.getProps(props, widgets, searchState, metadata);
+      return connectorDesc.getProps.call(this, props, widgets, searchState, metadata);
```

Comment:
There's only a single instance of every connectorDesc (connectSearchBox({connectorDesc})) at runtime. Is it a good idea to use context on it and allow storing state on them while it will be shared ultimately? In the connectInfiniteHits that should be OK since we do not expect more than one instance of it.

I am ok with the change, we just have to keep this in mind while we create connectors and allow people creating them.